utils/utils.c

```diff
@@ -178,12 +178,35 @@ int remove_directory(char *dirname)
 	return ret;
 }
 
+bool is_ancestor(char *path, char *data_path)
+{
+        int len = strlen(data_path);
+
+	if (data_path[len-1] == '/')
+		len--;
+
+	if (strncmp(path, data_path, len))
+		return false;
+
+	return (path[len] == '/' || path[len] == '\0');
+}
+
 int create_directory(char *dirname)
 {
 	int ret = -1;
 	char *oldname = NULL;
+	char *cwd_path = NULL;
+	char resolved_data_path[PATH_MAX];
+	char current_working_dir[PATH_MAX];
 
 	xasprintf(&oldname, "%s.old", dirname);
+	realpath(dirname, resolved_data_path);
+	cwd_path = getcwd(current_working_dir, PATH_MAX);
+
+	if (is_ancestor(cwd_path, resolved_data_path)) {
+		pr_err("Do not select the path which uftrace located\n"\
```

Comment:
`pr_err` adds error string based on the value of `errno` at the end.  In this case it's not a system error, it's an user's fault.

And cwd is not necessarily a directory of uftrace source/binary.  It's just a current working directory and uftrace should not remove it.  (i.e. do not run `rm -rf .` or `rm -rf ..`)